### PR TITLE
Introduce missing hits recovery in the `SiStripHitEfficiency` PCL workflow to be in synch with `CalibTree` based one

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This is a follow-up to PR https://github.com/cms-sw/cmssw/pull/41970. 
The goal of this PR is to extend the recovery of missing hits for the mkFit trajectories to the DQM-based `SiStripHitEfficiency` workflow (without this, the hit efficiency is overestimated in single-sided layers).

#### PR validation:

Run the following command:

```
cmsDriver.py stepReAlCa -s ALCA:PromptCalibProdSiStripHitEff --conditions 130X_dataRun3_Express_v2 --scenario pp --data --era Run3 --datatier ALCARECO --eventcontent ALCARECO --processName=ReAlCa -n 100000 --dasquery='file dataset=/StreamExpress/Run2023C-SiStripCalMinBias-Express-v4/ALCARECO run=368724' --nThreads=4
```

followed by:

```
cmsDriver.py stepHarvest -s ALCAHARVEST:SiStripHitEff --conditions 130X_dataRun3_Express_v2 --scenario pp --data --era Run3 --filein file:PromptCalibProdSiStripHitEff.root -n -1
```

and obtained the plot:

comparison

(as expected the efficiency for the DS layers gets reduced).

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but will be backported together with https://github.com/cms-sw/cmssw/pull/41970 .

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflows = 1001.2, 1001.3

**Comment by mmusich** [comment]
@cmsbuild, please test

File: CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyHarvester.cc [hunk 1]
```diff
@@ -401,8 +401,9 @@ void SiStripHitEfficiencyHarvester::dqmEndJob(DQMStore::IBooker& booker, DQMStor
             const auto num = h_module_found->getValue(det);
             const auto denom = h_module_total->getValue(det);
             if (denom) {
+              assert(num <= denom);  // can't have this happen
               const auto eff = num / denom;
-              const auto eff_up = TEfficiency::Bayesian(denom, num, .99, 1, 1, true);
+              const auto eff_up = (num <= denom) ? TEfficiency::Bayesian(denom, num, .99, 1, 1, true) : 1.f;
```
[the remark this answers is not in the document]
Comment (francescobrivio) [reply CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyHarvester.cc:406]: if this needs to be changes, could you also address https://github.com/cms-sw/cmssw/pull/41970#discussion_r1235288790? (It's absolutely not critical and can be posponed to further PR)

[the remark this answers is not in the document]
Comment (mmusich) [reply CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyHarvester.cc:406]: at this point it's pleonastic.

Comment (mmusich) [inline CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyHarvester.cc:406]:
```suggestion
              const auto eff_up = TEfficiency::Bayesian(denom, num, .99, 1, 1, true);
```

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
Failure is also in [IBs](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc11/CMSSW_13_2_X_2023-07-04-1100/pyRelValMatrixLogs/run/1001.3_RunSingleMuon2022B/step2_RunSingleMuon2022B.log#/).

**Comment by francescobrivio** [comment]
+1

 - failure in tests is unrelated to this PR, see https://github.com/cms-sw/cmssw/pull/42175#issuecomment-1620743786

**Comment by perrotta** [comment]
merge

**Pull request merged**